File: javap/OOPS/j_aggregation/book.java
package javap.OOPS.j_aggregation;

//Unidirectional relationship.
class Shape {
    int square(int a) {
        return a * a;
    }
}

class Geometry {
    Shape sq;

    int squareArea(int a) {
        sq = new Shape();
        return sq.square(a);
    }
}

class Operations {
    public static void main(String args[]) {
        Geometry g1 = new Geometry();
        System.out.println("Area :: " + g1.squareArea(2));
    }
}

File: javascriptp/OOPS/js_aggregation/book.js
var Square = function () {
  this.area = function (a) {
    return a * a;
  };
};

var Geometry = function (a) {
  var sq = new Square();
  this.area = function () {
    return sq.area(a);
  };
};

var g1 = new Geometry(2);
console.log("g1", g1.area());
